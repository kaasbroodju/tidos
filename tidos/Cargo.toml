[package]
name = "tidos"
version = { workspace = true }
authors = ["kaasbroodju"]
license = "MIT"
repository = "https://github.com/kaasbroodju/tidos"
description = "Tidos is a component framework that can be served by any web framework."
readme = "../README.md"
rust-version.workspace = true
edition.workspace = true
keywords = ["tidos", "web", "framework", "component"]
categories = ["web-programming"]

[package.metadata.docs.rs]
all-features = true

[dependencies]
rocket = { version = "0.5.1", optional = true }
tidos_macro = { workspace = true }

[dev-dependencies]
trybuild = "1.0.104"

[features]
rocket = ["dep:rocket"]
