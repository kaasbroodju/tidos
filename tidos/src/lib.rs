//! # Tidos - Documentation
//! Tidos is a high-performance Rust-based component framework that seamlessly integrates with any web framework, enabling developers to build dynamic web applications with ease. With Tidos’ powerful macros, you can intuitively create components directly within your Rust code. It even allows you to leverage Rust's pattern matching, loops, and conditionals inside your components—making your UI logic more expressive and maintainable.
//!
//! ## Usage
//! ```toml
//! tidos = "0.6.8"
//! ```
//! ## Example
//! ```rust,no_run
//! use tidos::{Component, Page};
//!
//! pub struct Greet {
//!     pub name: String,
//! }
//!
//! impl Component for Greet {
//!     fn to_render(&self, page: &mut Page) -> String {
//!         view! {
//!             <h1>Hello {&self.name}</h1>
//!         }
//!     }
//! }
//!
//!
//! // Example route from Rocket, but you can use any framework you want.
//! #[get("/")]
//! pub fn index() -> Page {
//!     page! {
//!         <main>
//!             <Greet name={ String::from("kaasbroodju") } />
//!         </main>
//!     }
//! }
//! //! ```

mod components;
mod page;

#[doc(hidden)]
pub mod internals;

pub use components::Component;
#[doc(hidden)]
pub use internals::sanitize;
pub use page::Page;
#[doc(inline)]
pub use tidos_macro::*;

#[doc(hidden)]
#[macro_export]
macro_rules! sanitize {
	($input:expr) => {
		&tidos::internals::sanitize(&$input)
	};
}
