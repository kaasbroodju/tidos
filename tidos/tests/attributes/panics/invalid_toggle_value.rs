use tidos::view;

fn toggle_attribute_implicit_variable_name() {
	assert_eq!(
		&view!{
			<input type="radio" name="day" value="monday" :checked="not a bool" />
		},
		r#"<input type="radio" name="day" value="monday" checked />"#
	)
}

fn main() {}