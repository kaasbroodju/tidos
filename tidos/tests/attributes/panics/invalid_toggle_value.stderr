error: Unable to have a toggle attribute from a literal, change it into the following:
       :checked
       :checked={ bool }
       checked="not a bool"
       checked={ "not a bool" }
 --> tests/attributes/panics/invalid_toggle_value.rs:6:59
  |
6 |             <input type="radio" name="day" value="monday" :checked="not a bool" />
  |                                                                    ^^^^^^^^^^^^
